src/openrct2/windows/view_clipping.c

```diff
@@ -0,0 +1,267 @@
+{
+	// Turn off view clipping when the window is closed.
+	rct_window *mainWindow = window_get_main();
+	if (mainWindow != NULL) {
+		mainWindow->viewport->flags &= ~VIEWPORT_FLAG_PAINT_CLIP_TO_HEIGHT;
+		window_invalidate(mainWindow);
+	}
+
+}
+
+static void window_view_clipping_close_button(rct_window *w)
+{
+	window_view_clipping_close();
+}
+
+static void window_view_clipping_mouseup(rct_window *w, int widgetIndex)
+{
+	rct_window *mainWindow;
+
+	// mouseup appears to be used for buttons, checkboxes
+	switch (widgetIndex) {
+	case WIDX_CLOSE:
+		window_close(w);
+		break;
+	case WIDX_CLIP_HEIGHT_CHECKBOX:
+		// Toggle height clipping.
+		mainWindow = window_get_main();
+		if (mainWindow != NULL) {
+			mainWindow->viewport->flags ^= VIEWPORT_FLAG_PAINT_CLIP_TO_HEIGHT;
+			window_invalidate(mainWindow);
+		}
+		window_invalidate(w);
+		break;
+	case WIDX_CLIP_HEIGHT_INCREASE:
+		if (gClipHeight < 255)
+			window_view_clipping_set_clipheight(w, gClipHeight + 1);
+		mainWindow = window_get_main();
+		if (mainWindow != NULL)
+			window_invalidate(mainWindow);
+		break;
+	case WIDX_CLIP_HEIGHT_DECREASE:
+		if(gClipHeight > 0)
+			window_view_clipping_set_clipheight(w, gClipHeight - 1);
+		mainWindow = window_get_main();
+		if (mainWindow != NULL)
+			window_invalidate(mainWindow);
+		break;
+	}
+}
+
+static void window_view_clipping_update(rct_window *w)
+{
+	const rct_widget *const widget = &window_view_clipping_widgets[WIDX_CLIP_HEIGHT_SLIDER];
+	const rct_scroll *const scroll = &w->scrolls[0];
+	const sint16 scroll_width = widget->right - widget->left - 1;
+	const uint8 clip_height = (uint8)(((float)scroll->h_left / (scroll->h_right - scroll_width)) * 255);
+	if (clip_height != gClipHeight) {
+		gClipHeight = clip_height;
+
+		// Update the main window accordingly.
+		rct_window *mainWindow = window_get_main();
+		if (mainWindow != NULL) {
+			window_invalidate(mainWindow);
+		}
+	}
+	widget_invalidate(w, WIDX_CLIP_HEIGHT_SLIDER);
+}
+
+static void window_view_clipping_invalidate(rct_window *w)
+{
+	colour_scheme_update(w);
+
+	widget_scroll_update_thumbs(w, WIDX_CLIP_HEIGHT_SLIDER);
+
+	rct_window *mainWindow = window_get_main();
+	if (mainWindow != NULL) {
+		widget_set_checkbox_value(w, WIDX_CLIP_HEIGHT_CHECKBOX, mainWindow->viewport->flags & VIEWPORT_FLAG_PAINT_CLIP_TO_HEIGHT);
+	}
+}
+
+static void window_view_clipping_paint(rct_window *w, rct_drawpixelinfo *dpi)
+{
+	window_draw_widgets(w, dpi);
+
+	// Clip height value
+	int x;
+	int y;
+	x = w->x + 8;
+	y = w->y + w->widgets[WIDX_CLIP_HEIGHT_VALUE].top;
+	gfx_draw_string_left(dpi, STR_VIEW_CLIPPING_HEIGHT_VALUE, NULL, w->colours[1], x, y);
+
+	x = w->x + w->widgets[WIDX_CLIP_HEIGHT_VALUE].left + 1;
+	y = w->y + w->widgets[WIDX_CLIP_HEIGHT_VALUE].top;
+	gfx_draw_string_left(dpi, STR_FORMAT_INTEGER, &gClipHeight, w->colours[1], x, y); //Printing the raw value.
+
+	// Print the value in the configured measurement units.
+	fixed16_1dp clipHeightValueInMeters;
```

Comment:
I tried with {LENGTH}.
It prints the correct measurement suffix (i.e., "ft" or "m"), but only appears to handle whole numbers, i.e. the decimal place in the fixed16_1dp is not shown.  Trying with a double just shows "0m".
I'd really like to have the decimal place for METRIC/SI units shown so the cut values match those shown in the viewport by the various "Height Marks" view options.